edx_when/models.py

```diff
@@ -75,6 +90,22 @@ def __str__(self):
         # TODO: return a string appropriate for the data fields
         return '<ContentDate, ID: {}>'.format(self.id)
 
+    def schedule_for_user(self, user):
+        """
+        Return the schedule for the supplied user that applies to this piece of content.
+        """
+        if isinstance(user, int):
```

Comment:
Is this a common pattern in platform code, to allow a user object or int?

Reply:
I wouldn't say it's common, and as I was thinking about how to justify the use here, I'm realizing that I don't think it actually helps (sometimes, it can help avoid unnecessary queries, IIRC, but I don't think that's true here).

Reply:
I right, I remember why this is a pattern now.

If the caller already has a user, and potentially might have prefetched the related enrollment and schedule, then it's better to take a user object, and use that prefetched data.

However, if the caller hasn't prefetched a user, then we don't actually need them to pass us a user object, it would be enough to just give us a user_id so that we can query for exactly the schedule we need.

Reply:
Sure, that makes sense - it just seems like an unusual pattern to take two entirely different types (beyond duck typing).

I'd expect either a docstring comment about that, having a separate method for each input type, or a pattern where there's an abstraction layer between those two paths (like a wrapper object for a user) or something.

Maybe just throw a docstring comment in there? Unless this is so common that all of platform does it and a docstring would be needless.